types/utils.d.ts

```diff
@@ -12,7 +12,7 @@ export type DeepPartial<T> = T extends Function
   type _DeepPartialArray<T> = Array<DeepPartial<T>>
 type _DeepPartialObject<T> = { [P in keyof T]?: DeepPartial<T[P]> };
 
-export type DistributiveArray<T> = T extends unknown ? T[] : never
+export type DistributiveArray<T> = [T] extends [unknown] ? Array<T> : never
```

Comment:
https://stackoverflow.com/questions/53996797/typescript-conditional-type-array-of-union-type-distribution